docs/pages/pmd/userdocs/pmd_report_formats.md

```diff
@@ -0,0 +1,339 @@
+tags: [pmd, userdocs]
+keywords: [formats, renderers]
+summary: "Overview of the built-in report formats for CPD"
+permalink: pmd_userdocs_report_formats.html
+author: Andreas Dangel <andreas.dangel@pmd-code.org>
+---
+
+## Overview
+
+PMD can report the found rule violations in various formats. Some formats can
+be customized further via properties. Violations might also be suppressed and there might
+be processing errors or configuration errors. Not all report formats display all information.
+
+The header of the sections below are used to select the format on the command line, as
+arguments to the `-format` option. When a format accepts *properties*,
+those can be specified with the `-property` / `-P` option on the command-line.
+
+{% include note.html content="Suppressed violations are only reported, if the CLI parameter `-showsuppressed` is set." %}
+
+## codeclimate
+
+Renderer for Code Climate JSON format.
+
+This format is used when running PMD within [Code Climate](https://codeclimate.com/).
+The renderer will stream JSON objects, each object is a own issue (a PMD rule violation). Each issue
+is separated by the null character (`\0`).
+
+The format is specified here: <https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md>.
+
+The code climate format doesn't support suppressed violations. It also doesn't report any errors. But it contains
+the full rule details for each reported rule violation.
+
+Example:
+
+```
+{"type":"issue","check_name":"GuardLogStatement","description":"Logger calls should be surrounded by log level guards.","content":{"body":"## GuardLogStatement\n\nSince: PMD 5.1.0\n\nPriority: Medium High\n\n[Categories](https://github.com/codeclimate/spec/blob/master/SPEC.md#categories): Style\n\n[Remediation Points](https://github.com/codeclimate/spec/blob/master/SPEC.md#remediation-points): 50000\n\nWhenever using a log level, one should check if the loglevel is actually enabled, or otherwise skip the associate String creation and manipulation.\n\n### Example:\n\n```java\n\n\n // Add this for performance\n if (log.isDebugEnabled() { ...\n log.debug('log something' + ' and ' + 'concat strings');\n\n \n``` \n\n### [PMD properties](https://pmd.github.io/pmd-6.22.0/pmd_devdocs_working_with_properties.html)\n\nName | Value | Description\n--- | --- | ---\nviolationSuppressRegex | | Suppress violations with messages matching a regular expression\nviolationSuppressXPath | | Suppress violations on nodes which match a given relative XPath expression.\nlogLevels | trace,debug,info,warn,error,log,finest,finer,fine,info,warning,severe | LogLevels to guard\nguardsMethods | isTraceEnabled,isDebugEnabled,isInfoEnabled,isWarnEnabled,isErrorEnabled,isLoggable | Method use to guard the log statement\n"},"categories":["Style"],"location":{"path":"/home/pmd/source/pmd-core/src/main/java/net/sourceforge/pmd/RuleContext.java","lines":{"begin":124,"end":125}},"severity":"normal","remediation_points":50000}
```

Comment:
The dead links checker reports the links
* `[PMD properties](https://pmd.github.io/pmd-6.22.0/pmd_devdocs_working_with_properties.html)`
* `[Categories](https://github.com/codeclimate/spec/blob/master/SPEC.md#categories)`

Looks like the renderer is out of date. I'll open an issue to track that